great_expectations/expectations/metrics/query_metrics/query_row_count.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Dict, Sequence, Union
+
+from great_expectations.compatibility.sqlalchemy import (
+    sqlalchemy as sa,
+)
+from great_expectations.core.metric_domain_types import MetricDomainTypes
+from great_expectations.execution_engine import (
+    SparkDFExecutionEngine,
+    SqlAlchemyExecutionEngine,
+)
+from great_expectations.expectations.metrics.metric_provider import metric_value
+from great_expectations.expectations.metrics.query_metric_provider import (
+    QueryMetricProvider,
+)
+
+if TYPE_CHECKING:
+    from great_expectations.compatibility import pyspark
+
+
+class QueryRowCount(QueryMetricProvider):
+    metric_name = "query.row_count"
+    value_keys = ("query",)
+
+    @metric_value(engine=SqlAlchemyExecutionEngine)
+    def _sqlalchemy(
```

Comment:
`UnexpectedRowsQuery` only supports SQL and Spark (see instruction #2 [here](https://docs.greatexpectations.io/docs/core/customize_expectations/use_sql_to_define_a_custom_expectation#procedure)). These 2 execution engines are all that has been implemented for `query.table` as well.